tools/analysis/analysis_runner.py

```diff
@@ -1,13 +1,51 @@
 # Analysis runner.
 import argparse
+import json
 import os
 import subprocess
+import urllib.parse
+import urllib.request
+from xml.etree import ElementTree
 
 _TOPIC = os.getenv(
     'OSSMALWARE_WORKER_TOPIC',
     'gcppubsub://projects/ossf-malware-analysis/topics/workers')
 _PACKAGES_BUCKET = os.getenv(
     'OSSF_MALWARE_ANALYSIS_PACKAGES', 'gs://ossf-malware-analysis-packages')
+_NPM_IGNORE_KEYS = ('modified', 'created')
+
+
+def _npm_versions_for_package(pkg):
+  safe_pkg = urllib.parse.quote_plus(pkg)
+  url = f'https://registry.npmjs.com/{safe_pkg}'
+  resp = urllib.request.urlopen(url)
+  data = json.loads(resp.read())
+  versions = data.get('time', {}).keys()
+  return [v for v in versions if v not in _NPM_IGNORE_KEYS][::-1]
+
+
+def _pypi_versions_for_package(pkg):
+  safe_pkg = urllib.parse.quote_plus(pkg)
+  url = f'https://pypi.org/rss/project/{safe_pkg}/releases.xml'
```

Comment:
nit: The Python JSON API is much simpler and doesn't require XML:

e.g. OSV uses this for its own version enumeration:

https://github.com/google/osv/blob/737948b5060efeffb1cf47fe085d187a3d0304a4/lib/osv/ecosystems.py#L126

Reply:
That API is much better. Thanks!